pkg/certificate/interface.go

```diff
@@ -0,0 +1,101 @@
+package certificate
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+type ProviderType string
+
+const (
+	Auto        ProviderType = "auto"
+	CertManager ProviderType = "cert-manager"
+	// add more ...
+)
+
+func NewProvider(pt ProviderType) Provider {
+	switch pt {
+	case Auto:
+		return nil // change me later
+	case CertManager:
+		return nil // change me later
+	}
+	panic(fmt.Sprintf("unknown provider type: %s", pt))
+}
+
+type Config struct {
+	CommonName       string
+	Organization     []string
+	DNSNames         []string
+	ValidityDuration time.Duration
+	CA               []byte
+
+	// ExtraConfig contains provider specific configurations.
+	ExtraConfig map[string]string
+}
+
+type Provider interface {
+	// EnsureCertificateSecret ensures the specified certificate is
+	// available as a Secret in Kubernetes. If the Secret does not
+	// exist, it will be created.
+	//
+	// Parameters:
+	// - ctx: Context for cancellation and deadlines.
+	// - secretName: Name of the Secret to ensure.
+	// - namespace: Namespace where the Secret should reside.
+	// - cfg: Configuration for the certificate.
+	//
+	// Returns:
+	// - nil if the operation succeeds, or an error otherwise.
+	EnsureCertificateSecret(ctx context.Context, secretName string, namespace string, cfg Config) error
```

Comment:
cfg can be a pointer in all interface functions.

Reply:
updated & thx